### PR TITLE
Adds dialog before sign in prompt

### PR DESCRIPTION
**Pull request by protocol86**
Adds dialog before prompt.  Localization for email/password dialog will be added in a follow-up PR.

### CODE REVIEW

File: FirebaseAuthUI/FUIAuth.m [hunk 1]
```diff
@@ -309,12 +311,24 @@ - (void)signInWithEmailHint:(NSString *)emailHint
...
       if ([existingFederatedProviderID isEqualToString:FIREmailAuthProviderID]) {
-        FUIAuth *authUI = [[FUIAuth alloc]initWithAuth:tempAuth];
-        // Email password sign-in
-        FUIPasswordSignInViewController *controller =
-            [[FUIPasswordSignInViewController alloc] initWithAuthUI:authUI email:emailHint];
-        controller.onDismissCallback = completion;
-        [presentingViewController pushViewController:controller];
+
+        [FUIAuthBaseViewController showSignInAlertWithEmail:emailHint
+                                          providerShortName:@"Email/Password"
```
Comment (morganchen12) [inline FirebaseAuthUI/FUIAuth.m]: These strings will eventually have to be localized (can be in a follow up PR).

File: FirebaseAuthUI/FUIAuth.m [hunk 2]
```diff
@@ -309,12 +311,24 @@ - (void)signInWithEmailHint:(NSString *)emailHint
...
+                                        providerSignInLabel:@"Sign in with Email/Password"
+                                   presentingViewController:presentingViewController
+                                              signinHandler:^{
+          FUIAuth *authUI = [[FUIAuth alloc]initWithAuth:tempAuth];
```
Comment (morganchen12) [inline FirebaseAuthUI/FUIAuth.m]: nit: space between `alloc]` and `initWithAuth`

File: FirebaseAuthUI/FUIAuth.m [hunk 3]
```diff
@@ -324,20 +338,32 @@ - (void)signInWithEmailHint:(NSString *)emailHint
...
+          }];
+        }
+                                              cancelHandler:^{
+          if (completion) {
+            completion(nil, originalError);
           }
-          [tempAuth signInAndRetrieveDataWithCredential:credential completion:completion];
-        }];
+         }
+        ];
```
Comment (morganchen12) [inline FirebaseAuthUI/FUIAuth.m]: nit: these should probably be on the same line

File: FirebaseAuthUI/FUIAuth.m (hunk 1 above)
Comment (protocol86) [reply FirebaseAuthUI/FUIAuth.m]: Will do, had added as part of the PR description :)

File: FirebaseAuthUI/FUIAuth.m (hunk 3 above)
Comment (protocol86) [reply FirebaseAuthUI/FUIAuth.m]: Done, thanks!

**Pull request merged**